### PR TITLE
Track overview balance

### LINKED ISSUE #5880: Ability to measure view and user interaction with the Balance card on Payments > Overview page
We need to be able to:
- Count how many times user view the card. We might be able to use Payments > Overview page view ([read](https://github.com/Automattic/woocommerce-payments/issues/5771#issuecomment-1468699248)).

- Count user's interaction with the currency tab.

224581592-419d1027-d343-4ccd-88a0-539c4219a8f9

- Count user's interaction, clicking, to open and close the tooltip.

224232319-3dd3bb14-b0c8-469f-8fdc-8557fcc82199

### PR DESCRIPTION
**Pull request by shendy-a8c**
Fixes https://github.com/Automattic/woocommerce-payments/issues/5880

#### Changes proposed in this Pull Request

- Track when user selects currency tab by click or by keyboard `Enter` key.
- Track when user opens the balance tooltips by click or by keyboard `Enter` key.

ps: The issue https://github.com/Automattic/woocommerce-payments/issues/5880 mentions to track when tooltip is closed but I think it's not too important to track it, so I do not include it in this PR. Let me know if I should.

#### Testing instructions

* Make sure wp-admin > WooCommerce > Settings > Advanced > WooCommerce.com > `Enable tracking` is on.

* Select a currency tab by clicking or by navigating using the `tab` key and hit the `Enter`.
* Expect a request to `t.gif` with payload consisting of `_en: wcadmin_wcpay_overview_balance_currency_click` and `currency: {currency}`.

* Open the balance tooltips by clicking or by navigating using the `tab` key and hit the `Enter` key the currency tab.
* Expect a request to `t.gif` with payload consisting of `_en: wcadmin_wcpay_overview_balance_tooltip_click ` and `context: {tooltip label}`.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6307` or branch name `add/5880-track-overview-balance` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5880-track-overview-balance) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: a71a62d3eba797bdb234755516e22f454f759976
- Build time: 2023-05-16 23:17:06 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/add-5880-track-overview-balance [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: add
```
Comment (haszari) [inline changelog/add-5880-track-overview-balance]: This is `dev` – not adding anything that benefits merchants.

File: client/components/account-balances/balance-tooltip.tsx [hunk 2]
```diff
@@ -18,10 +19,20 @@ const BalanceTooltip: React.FC< BalanceTooltipProps > = ( {
 	label,
 	content,
 } ) => {
+	const onClick = () => {
+		wcpayTracks.recordEvent(
+			wcpayTracks.events.OVERVIEW_BALANCE_TOOLTIP_CLICK,
```
Comment (haszari) [inline client/components/account-balances/balance-tooltip.tsx]: I don't know if we need to add events for tooltip clicks - we've been focusing on the core views and interactions, and didn't add view or click events for the various notices.

File: client/components/account-balances/balances-tab-panel.tsx [hunk 3]
```diff
@@ -57,6 +58,12 @@ const AccountBalancesTabPanel: React.FC = () => {
 
 	const onTabSelect = ( tabName: BalanceTab[ 'name' ] ) => {
 		setSelectedCurrency( tabName );
+		wcpayTracks.recordEvent(
+			wcpayTracks.events.OVERVIEW_BALANCE_CURRENCY_CLICK,
+			{
+				currency: tabName,
```
Comment (haszari) [inline client/components/account-balances/balances-tab-panel.tsx]: Suggest wording the action and the property so it's 100% clear that the `currency` is the newly chosen currency, e.g. `wcadmin_wcpay_overview_balances_currency_tab_click selected_currency=USD|EUR|GBP` etc.

File: client/components/account-balances/balance-tooltip.tsx (hunk 2 above)
Comment (haszari) [reply client/components/account-balances/balance-tooltip.tsx]: To clarify, I suggest removing this event (and supporting code) unless it's important/required by product team to track interactions with the tooltip(s).

File: changelog/add-5880-track-overview-balance (hunk 1 above)
Comment (shendy-a8c) [reply changelog/add-5880-track-overview-balance]: Good point. Addressed [54c7942](https://github.com/Automattic/woocommerce-payments/pull/6307/commits/54c79428d719758ead0f4983884281c834d67010).

File: client/components/account-balances/balance-tooltip.tsx (hunk 2 above)
Comment (shendy-a8c) [reply client/components/account-balances/balance-tooltip.tsx]: Even better. Making this PR much simpler. Addressed here [ca88de8](https://github.com/Automattic/woocommerce-payments/pull/6307/commits/ca88de804a1fd82c607b62f6b66044daccb02aa4).

File: client/components/account-balances/balances-tab-panel.tsx (hunk 3 above)
Comment (shendy-a8c) [reply client/components/account-balances/balances-tab-panel.tsx]: Addressed here [1dcd9e6](https://github.com/Automattic/woocommerce-payments/pull/6307/commits/1dcd9e602f92ae3ba7424c223455e4ef693b981c).

File: client/tracks/index.js [hunk 4]
```diff
@@ -62,6 +62,8 @@ const events = {
 	CONNECT_ACCOUNT_KYC_MODAL_OPENED: 'wcpay_connect_account_kyc_modal_opened',
 	DEPOSITS_ROW_CLICK: 'wcpay_deposits_row_click',
 	DEPOSITS_DOWNLOAD_CSV_CLICK: 'wcpay_deposits_download',
+	OVERVIEW_BALANCE_CURRENCY_CLICK:
+		'wcpay_overview_balances_currency_tab_click',
```
Comment (Jinksi) [inline client/tracks/index.js]: Nit/minor suggestion: use `OVERVIEW_BALANCES` naming to remain consistent with event name (`overview_balances`).
```suggestion
	OVERVIEW_BALANCES_CURRENCY_CLICK:
		'wcpay_overview_balances_currency_tab_click',
```

**Review by Jinksi — APPROVED** [review]
Nice one @shendy-a8c 🙌 

I've tested with a multi-currency store, selecting between account balance currency tabs via both keyboard and mouse click results in the tracks events being recorded with the correct `selected_currency` event prop.

I've left a minor suggestion, feel free to take it or ~leave it~ 👍

**Update**: I've committed this suggestion in the interest of getting this shipped 🚀

**Comment by Jinksi** [comment]
In the interest of getting this shipped, I'll merged this on your behalf @shendy-a8c 🚀

**Pull request merged**